### PR TITLE
osd/PG: Remove unnecessary assertion

### PR DESCRIPTION
**Pull request by shinobu-x**
Signed-off-by: Shinobu Kinjo <shinobu@redhat.com>

### CODE REVIEW

File: src/osd/PG.cc [hunk 1]
```diff
@@ -818,7 +818,6 @@ void PG::check_past_interval_bounds() const
       derr << info.pgid << " required past_interval bounds are"
 	   << " not empty [" << rpib << ") but past_intervals "
 	   << past_intervals << " is empty" << dendl;
-      assert(!past_intervals.empty());
```
Comment (xiexingguo) [inline src/osd/PG.cc]: I think this assert is intentional made to catch some unexpected/bad behaviour..

**Comment by shinobu-x** [comment]
test this please

**Comment by shinobu-x** [comment]
test this please

File: src/osd/PG.cc (hunk 1 above)
Comment (shinobu-x) [reply src/osd/PG.cc]: In that case, it may probably make more sense to leave some comment.

Comment (liewegas) [reply src/osd/PG.cc]: the derr and error() lines above are the comment

**Pull request closed without merging**